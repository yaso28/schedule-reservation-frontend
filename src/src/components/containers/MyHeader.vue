<template>
  <CHeader>
    <!-- sidebar toggler begin -->
    <CToggler
      id="toggle-sidebar-mobile"
      in-header
      class="ml-3 d-lg-none"
      @click="toggleMobile"
    />
    <CToggler
      id="toggle-sidebar-desktop"
      in-header
      class="ml-3 d-md-down-none"
      @click="toggleDesktop"
    />
    <!-- sidebar toggler end -->

    <!-- usermenu dropdown begin -->
    <CHeaderNav class="mr-3 ml-auto">
      <CDropdown
        id="usermenu-dropdown"
        in-nav
        class="c-header-nav-items"
        placement="bottom-end"
        :toggler-text="usermenuText"
        add-toggler-classes="c-header-nav-link"
        add-menu-classes="pt-0"
      >
        <template v-if="isLogin">
          <CDropdownItem :to="{ name: 'logout' }">
            {{ $t('action.logout') }}
          </CDropdownItem>
        </template>
        <template v-else>
          <CDropdownItem :to="{ name: 'login' }">
            {{ $t('action.login') }}
          </CDropdownItem>
        </template>
      </CDropdown>
    </CHeaderNav>
    <!-- usermenu dropdown end -->
  </CHeader>
</template>

<script>
import { mapState, mapGetters } from 'vuex'

export default {
  name: 'MyHeader',
  computed: {
    ...mapState({
      user: state => state.auth.user
    }),
    ...mapGetters({
      isLogin: 'auth/isLogin'
    }),
    usermenuText () {
      return this.isLogin ? this.user.name : this.$t('user.guest')
    }
  },
  methods: {
    toggleMobile () {
      this.$store.commit('sidebar/toggleMobile')
    },
    toggleDesktop () {
      this.$store.commit('sidebar/toggleDesktop')
    }
  }
}
</script>
