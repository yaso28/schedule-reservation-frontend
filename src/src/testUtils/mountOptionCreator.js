import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
