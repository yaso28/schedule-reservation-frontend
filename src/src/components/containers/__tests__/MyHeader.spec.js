import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MyHeader from '../MyHeader'
import storeModules from '@/store/modules'

jest.mock('@/store/modules')

describe('MyHeader', () => {
  let mountOption
  let wrapper
  let store
  beforeEach(() => {
    mountOption = mountOptionCreator.create(storeModules)
    store = mountOption.store
    store.state.auth.user = null
    storeModules.auth.getters.isLogin.mockReturnValue(false)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('認証なし', () => {
    beforeEach(() => {
      wrapper = mount(MyHeader, mountOption)
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('認証あり', () => {
    beforeEach(() => {
      store.state.auth.user = {
        id: 2,
        name: 'やそやそ',
        email: 'yaso@sample.com'
      }
      storeModules.auth.getters.isLogin.mockReturnValue(true)
      wrapper = mount(MyHeader, mountOption)
    })

    test('ユーザーメニュードロップダウン表示', () => {
      expect(wrapper.find('#usermenu-dropdown').element).toMatchSnapshot()
    })
  })

  describe('トグルボタン（モバイル）クリック', () => {
    beforeEach(() => {
      wrapper.find('#toggle-sidebar-mobile').trigger('click')
    })

    test('サイドバートグル（モバイル）呼出', () => {
      expect(storeModules.sidebar.mutations.toggleMobile).toHaveBeenCalled()
    })
  })

  describe('トグルボタン（デスクトップ）クリック', () => {
    beforeEach(() => {
      wrapper.find('#toggle-sidebar-desktop').trigger('click')
    })

    test('サイドバートグル（デスクトップ）呼出', () => {
      expect(storeModules.sidebar.mutations.toggleDesktop).toHaveBeenCalled()
    })
  })
})
