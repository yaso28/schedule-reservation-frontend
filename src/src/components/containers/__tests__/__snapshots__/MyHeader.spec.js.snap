// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyHeader 認証あり ユーザーメニュードロップダウン表示 1`] = `
<li
  class="c-header-nav-items dropdown nav-item"
  id="usermenu-dropdown"
>
  <a
    aria-expanded="false"
    aria-haspopup="true"
    class="c-header-nav-link nav-link dropdown-toggle"
    href="#"
    target="_self"
  >
    やそやそ
  </a>
  <!---->
  <div
    class="pt-0 dropdown-menu"
  >
    <a
      class="dropdown-item"
      href="/logout"
      role="menuitem"
      target="_self"
    >
      
          ログアウト
        
    </a>
  </div>
</li>
`;

exports[`MyHeader 認証なし 表示 1`] = `
<header
  class="c-header c-header-light c-header-fixed"
>
  <button
    class="ml-3 d-lg-none c-header-toggler"
    id="toggle-sidebar-mobile"
    type="button"
  >
    <span
      class="c-header-toggler-icon"
    />
  </button>
   
  <button
    class="ml-3 d-md-down-none c-header-toggler"
    id="toggle-sidebar-desktop"
    type="button"
  >
    <span
      class="c-header-toggler-icon"
    />
  </button>
   
  <ul
    class="c-header-nav mr-3 ml-auto"
  >
    <li
      class="c-header-nav-items dropdown nav-item"
      id="usermenu-dropdown"
    >
      <a
        aria-expanded="false"
        aria-haspopup="true"
        class="c-header-nav-link nav-link dropdown-toggle"
        href="#"
        target="_self"
      >
        ゲスト
      </a>
      <!---->
      <div
        class="pt-0 dropdown-menu"
      >
        <a
          class="dropdown-item"
          href="/lv2au5ldfg-login"
          role="menuitem"
          target="_self"
        >
          
          ログイン
        
        </a>
      </div>
    </li>
  </ul>
</header>
`;
